website/scripts/convert_ipynb_to_mdx.py

```diff
@@ -30,526 +31,779 @@
+    """
+    Handle how to plain cell output should be displayed in MDX.
+
+    Args:
+        values (List[Dict[int, NotebookNode]]): Bokeh tagged cell outputs.
+
+    Returns:
+        List[Tuple[int, str]]: A list of tuples, where the first entry in the tuple is
+            the index where the output occurred from the cell, and the second entry of
+            the tuple is the MDX formatted string.
+    """
+    output = []
+    for value in values:
+        index = value["index"]
+        data = value["data"]
+        data = [line.strip() for line in data.splitlines() if line]
+        data = [datum for datum in data if datum]
+        if data:
+            data = "\n".join([line for line in value["data"].splitlines() if line])
+            output.append(
+                (index, f"<CellOutput>\n{{\n  `{data}`\n}}\n</CellOutput>\n\n")
+            )
+    return output
+
+
+def handle_plotly(
+    values: List[Dict[str, NotebookNode]],
+    plot_data_folder: PathLike,
+) -> List[Tuple[int, str]]:
+    """
+    Convert Plotly outputs to MDX.
+
+    Args:
+        values (List[Dict[int, NotebookNode]]): Bokeh tagged cell outputs.
+        plot_data_folder (PathLike): Path to the folder where plot data should be
+            stored.
+
+    Returns:
+        List[Tuple[int, str]]: A list of tuples, where the first entry in the tuple is
+            the index where the output occurred from the cell, and the second entry of
+            the tuple is the MDX formatted string.
+    """
+    output = []
+    for value in values:
+        index = value["index"]
+        data = value["data"]
+        file_name = str(uuid.uuid4())
+        file_path = str(plot_data_folder / f"{file_name}.json")
+        path_to_data = f"./assets/plot_data/{file_name}.json"
+        output.append(
+            (index, f"<PlotlyFigure data={{require('{path_to_data}')}} />\n\n")
+        )
+        with open(file_path, "w") as f:
+            json.dump(data, f, indent=2)
+    return output
+
+
+def handle_tqdm(values: List[Dict[int, NotebookNode]]) -> List[Tuple[int, str]]:
+    """
+    Handle the output of tqdm.
+
+    tqdm will be displayed as separate CellOutput React components if we do not
+    aggregate them all into a single CellOutput object, which is what this method does.
+
+    Args:
+        values (List[Dict[int, NotebookNode]]): Bokeh tagged cell outputs.
+
+    Returns:
+        List[Tuple[int, str]]: A list of tuples, where the first entry in the tuple is
+            the index where the output occurred from the cell, and the second entry of
+            the tuple is the MDX formatted string.
+    """
+    output = sorted(values, key=lambda item: item["index"])
+    index = output[0]["index"]
+    md = "\n".join([item["data"] for item in output if item["data"]])
+    return [(index, f"<CellOutput>\n{{\n  `{md}`\n}}\n</CellOutput>\n\n")]
 
-    # Begin to build the mdx string.
+
+def handle_cell_outputs(  # noqa: C901 (flake8 too complex)
```

Comment:
Our internal linter still insist that this function is too complex. Can we break it down into a few smaller ones?